security-analysis/security-analysis-api/src/main/java/com/powsybl/security/SecurityAnalysisResultBuilder.java

```diff
@@ -31,6 +32,7 @@ public class SecurityAnalysisResultBuilder {
     // Below are volatile objects used for building the actual complete result
     private final PreContingencyResult preContingencyResult;
     private final List<PostContingencyResult> postContingencyResults = Collections.synchronizedList(new ArrayList<>());
+    private final Map<String, ScenarioResult> scenarioResults = Collections.synchronizedMap(new HashMap<>());
```

Comment:
Better to use a list like for post-contingencies, otherwise the sort order will be random.